src/meltano/core/settings_store.py

```diff
@@ -50,6 +50,28 @@ def __str__(self) -> str:
         return f"Conflicting values for setting found in: {self.setting_names}"
 
 
+class MultipleEnvVarsSetException(Exception):
+    """Occurs when a setting value is set via multiple environment variable names."""
+
+    def __init__(self, names):
+        """Instantiate the error.
+
+        Args:
+            setting_names: the name/aliases where conflicting values are set
+
+        """
+        self.names = names
+        super().__init__(names)
+
+    def __str__(self) -> str:
+        """Represent the error as a string.
+
+        Returns:
+            string representation of the error
+        """
+        return f"Setting value set via multiple environment variables: {self.names}"
```

Comment:
Read on its own, I wonder if a user might not immediately understand the context. 

Should we have the word "error"?:

> Error: Setting value set via multiple environment variables: ..."

or a normative assertion or complaint could work here too:

> Setting value cannot not be determined due to conflicting environment variables: ..."

Reply:
Good point, I've updated it to add the word "Error".